src/connection.js

```diff
@@ -1057,6 +1101,41 @@ class Connection extends EventEmitter {
     }
   }
 
+  processLogin7KerberosResponse() {
+    if (this.loggedIn) {
+      return this.dispatchEvent('loggedIn');
+    }
+    else if (this.ntlmpacket && this.kerberos) {
+      this.sspiClientResponsePending = true;
+      this.kerberos.authGSSClientStep(this.context,
+        this.ntlmpacketBuffer.toString('base64', 0, this.ntlmpacketBuffer.length), (err, result) => {
+          if (err) {
+            this.emit('error', new Error(err.toString()));
+            return this.close();
+          }
+
+          //verify if kerberos auth was successful, ie, GSS_C_COMPLETE flag returned
+          if (!((null != result) && ('number' === typeof (result)) && (1 === result /* GSS_C_COMPLETE */))) {
+            this.emit('error', new Error('Expected GSS_C_COMPLETE flag not received, kerberos authentication failed'));
+            return this.close();
+          }
+
+          this.sspiClientResponsePending = false;
+
+          // clear the ntlmpacket
+          delete this.ntlmpacketBuffer;
```

Comment:
Deleting properties has negative performance implications in V8, if I remember correctly. 🤔

Reply:
sure, I'll double check that.